lib/dependencies/ContextDependencyHelpers.js

```diff
@@ -5,19 +5,43 @@
 var ContextDependencyHelpers = exports;
 
 ContextDependencyHelpers.create = function(Dep, range, param, expr, options) {
-	var dep;
-	if(param.isWrapped() && (param.prefix && param.prefix.isString() || param.postfix && param.postfix.isString())) {
-		var prefix = param.prefix && param.prefix.isString() ? param.prefix.string : "";
-		var postfix = param.postfix && param.postfix.isString() ? param.postfix.string : "";
-		var prefixRange = param.prefix && param.prefix.isString() ? param.prefix.range : null;
-		var valueRange = [prefixRange ? prefixRange[1] : param.range[0], param.range[1]];
-		var idx = prefix.lastIndexOf("/");
-		var context = ".";
+	var dep, prefix, postfix, prefixRange, valueRange, idx, context, regExp;
+	if(param.isTemplateString()) {
+		prefix = param.quasis[0].string;
+		postfix = param.quasis.length > 1 ? param.quasis[param.quasis.length - 1].string : "";
+		prefixRange = [param.quasis[0].range[0], param.quasis[0].range[1]];
+		valueRange = param.range;
+		idx = prefix.lastIndexOf("/");
+		context = ".";
 		if(idx >= 0) {
 			context = prefix.substr(0, idx);
 			prefix = "." + prefix.substr(idx);
 		}
-		var regExp = new RegExp("^" +
+		// If there are more than two quasis, maybe the generated RegExp can be more precise?
+		regExp = new RegExp("^" +
+			prefix.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "\\$&") +
+			options.wrappedContextRegExp.source +
+			postfix.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "\\$&") + "$");
+		dep = new Dep(context, options.wrappedContextRecursive, regExp, range, valueRange);
+		dep.loc = expr.loc;
+		dep.replaces = [{
+			range: prefixRange,
+			value: prefix
+		}];
+		dep.critical = options.wrappedContextCritical && "a part of the request of a dependency is an expression";
+		return dep;
+	} else if(param.isWrapped() && (param.prefix && param.prefix.isString() || param.postfix && param.postfix.isString())) {
+		prefix = param.prefix && param.prefix.isString() ? param.prefix.string : "";
+		postfix = param.postfix && param.postfix.isString() ? param.postfix.string : "";
+		prefixRange = param.prefix && param.prefix.isString() ? param.prefix.range : null;
+		valueRange = [prefixRange ? prefixRange[1] : param.range[0], param.range[1]];
+		idx = prefix.lastIndexOf("/");
+		context = ".";
+		if(idx >= 0) {
+			context = prefix.substr(0, idx);
+			prefix = "." + prefix.substr(idx);
+		}
+		regExp = new RegExp("^" +
```

Comment:
All the `var` declarations were hoisted to the top of the function as they were duplicated in the other branch.